back/app/services/projects_finder_admin_service.rb

```diff
@@ -117,23 +117,43 @@ def self.search(scope, params = {})
   end
 
   def self.filter_date(scope, params = {})
-    start_at = params[:start_at]
-    end_at = params[:end_at]
-    return scope if start_at.blank? && end_at.blank?
+    raw_start = params[:start_at]
+    raw_end = params[:end_at]
+    return scope if raw_start.blank? && raw_end.blank?
 
-    start_at ||= Date.new(1970, 1, 1)
-    end_at ||= DateTime::Infinity
+    start_at = parse_date(raw_start)
+    end_at   = parse_date(raw_end)
 
-    overlapping_project_ids = Phase
-      .select(:project_id)
-      .where(
+    query_start_at = start_at || Date.new(1970, 1, 1)
+
+    overlapping_project_ids = if end_at.present?
+      Phase.select(:project_id).where(
         "(start_at, coalesce(end_at, 'infinity'::DATE)) OVERLAPS (?, ?)",
-        start_at, end_at
+        query_start_at,
+        end_at
+      )
+    else
+      Phase.select(:project_id).where(
+        "(start_at, coalesce(end_at, 'infinity'::DATE)) OVERLAPS (?, 'infinity')",
+        query_start_at
       )
+    end
 
     scope.where(id: overlapping_project_ids)
   end
 
+  def self.parse_date(date_input)
```

Comment:
I think we can replace this by `TimeBoundariesParser` (`back/lib/time_boundaries_parser.rb`), which does something very similar, and which is what we usually use in this situation. Might need to update some unit tests after that though, since it relies on certain app config settings- e.g. you might need to do `AppConfiguration.instance.update!(created_at: Date.new(2020, 1, 1))` in a `before` block (see e.g. `back/engines/commercial/report_builder/spec/services/report_builder/queries/demographics_spec.rb`)